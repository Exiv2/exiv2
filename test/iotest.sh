#!/usr/bin/env bash
# Test driver for file i/o

source ./functions.source

(   cd "$testdir"

    errors=0
    test_files="table.jpg smiley2.jpg ext.dat"
    echo
    printf "file io tests"
    copyTestFiles $test_files
    for i in $test_files; do runTest iotest $i s1 s2; done

    printf "\n---------------------------------------------------------\n"
    if [ $errors -eq 0 ]; then
       echo 'All test cases passed'
    else
       echo $errors 'test case(s) failed!'
    fi
)

sniff() {
    # Format spec for stat
    F='-c%s'
    if [ $PLATFORM  == 'Darwin' -o $PLATFORM == 'NetBSD' -o $PLATFORM == 'FreeBSD' ]; then
        F='-f%z'
    fi
    echo $(stat $F s0 s1 s2 ../data/table.jpg) $(checkSum s0) $(checkSum s1) $(checkSum s2) $(checkSum ../data/table.jpg)
}

# Test http I/O
if [ "$EXIV2_PORT" != "None" -a "$EXIV2_HTTP" != "None" ]; then
    startHttpServer
    if [ ! -z $exiv2_httpServer ]; then
        (   cd "${testdir}" 
            >&2 printf "*** HTTP tests begin\n"

            cd "$testdir"
            test_files="table.jpg Reagan.tiff exiv2-bug922a.jpg"
            for i in $test_files; do
                runTest iotest s0 s1 s2 $exiv2_url/data/$i
                for t in s0 s1 s2       $exiv2_url/data/$i; do
                    runTest exiv2 -g City -g DateTime $t
                done
            done

            runTest iotest s0 s1 s2 $exiv2_url/data/table.jpg 0    ; sniff
            runTest iotest s0 s1 s2 $exiv2_url/data/table.jpg 10   ; sniff
            runTest iotest s0 s1 s2 $exiv2_url/data/table.jpg 1000 ; sniff

            >&2 printf "*** HTTP tests end\n"
        ) > $results
        reportTest
    fi
    closeHttpServer
fi

# That's all Folks!
##