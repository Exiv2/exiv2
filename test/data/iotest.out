Exif.Image.DateTime                          Ascii      21  2004-07-13T21:23:44Z
Iptc.Application2.City                       String      7  Seattle
Exif.Image.DateTime                          Ascii      21  2004-07-13T21:23:44Z
Iptc.Application2.City                       String      7  Seattle
Exif.Image.DateTime                          Ascii      21  2004-07-13T21:23:44Z
Iptc.Application2.City                       String      7  Seattle
Exif.Image.DateTime                          Ascii      21  2004-07-13T21:23:44Z
Iptc.Application2.City                       String      7  Seattle
Exif.Image.DateTime                          Ascii      20  2012:04:07 16:11:27
Exif.Photo.DateTimeOriginal                  Ascii      20  2004:06:21 23:37:53
Exif.Photo.DateTimeDigitized                 Ascii      20  2004:06:21 23:37:53
Iptc.Application2.City                       String     19  Straits of Magellan
Xmp.photoshop.City                           XmpText    19  Straits of Magellan
Exif.Image.DateTime                          Ascii      20  2012:04:07 16:11:27
Exif.Photo.DateTimeOriginal                  Ascii      20  2004:06:21 23:37:53
Exif.Photo.DateTimeDigitized                 Ascii      20  2004:06:21 23:37:53
Iptc.Application2.City                       String     19  Straits of Magellan
Xmp.photoshop.City                           XmpText    19  Straits of Magellan
Exif.Image.DateTime                          Ascii      20  2012:04:07 16:11:27
Exif.Photo.DateTimeOriginal                  Ascii      20  2004:06:21 23:37:53
Exif.Photo.DateTimeDigitized                 Ascii      20  2004:06:21 23:37:53
Iptc.Application2.City                       String     19  Straits of Magellan
Xmp.photoshop.City                           XmpText    19  Straits of Magellan
Exif.Image.DateTime                          Ascii      20  2012:04:07 16:11:27
Exif.Photo.DateTimeOriginal                  Ascii      20  2004:06:21 23:37:53
Exif.Photo.DateTimeDigitized                 Ascii      20  2004:06:21 23:37:53
Iptc.Application2.City                       String     19  Straits of Magellan
Xmp.photoshop.City                           XmpText    19  Straits of Magellan
Exif.Image.DateTime                          Ascii      20  2015:02:13 20:46:51
Exif.Photo.DateTimeOriginal                  Ascii      20  2015:02:13 13:51:35
Exif.Photo.DateTimeDigitized                 Ascii      20  2015:02:13 13:51:35
Exif.Image.DateTime                          Ascii      20  2015:02:13 20:46:51
Exif.Photo.DateTimeOriginal                  Ascii      20  2015:02:13 13:51:35
Exif.Photo.DateTimeDigitized                 Ascii      20  2015:02:13 13:51:35
Exif.Image.DateTime                          Ascii      20  2015:02:13 20:46:51
Exif.Photo.DateTimeOriginal                  Ascii      20  2015:02:13 13:51:35
Exif.Photo.DateTimeDigitized                 Ascii      20  2015:02:13 13:51:35
Exif.Image.DateTime                          Ascii      20  2015:02:13 20:46:51
Exif.Photo.DateTimeOriginal                  Ascii      20  2015:02:13 13:51:35
Exif.Photo.DateTimeDigitized                 Ascii      20  2015:02:13 13:51:35
568 568 568 568 c245b8764e4cf1104374787e21b6ef63 c245b8764e4cf1104374787e21b6ef63 c245b8764e4cf1104374787e21b6ef63 c245b8764e4cf1104374787e21b6ef63
568 568 568 568 c245b8764e4cf1104374787e21b6ef63 c245b8764e4cf1104374787e21b6ef63 c245b8764e4cf1104374787e21b6ef63 c245b8764e4cf1104374787e21b6ef63
568 568 568 568 c245b8764e4cf1104374787e21b6ef63 c245b8764e4cf1104374787e21b6ef63 c245b8764e4cf1104374787e21b6ef63 c245b8764e4cf1104374787e21b6ef63
